### PR TITLE
feat(world-local): add baseUrl config and improve port handling

### PR DESCRIPTION
**Pull request by aryasaatvik**
## Add baseUrl configuration and improve port handling

This PR adds baseUrl configuration support to enable HTTPS and custom hostnames in local development, while maintaining automatic port detection as a fallback.

### Key Features

**BaseUrl override:**
- Configure via `baseUrl` config option or `WORKFLOW_EMBEDDED_BASE_URL` env var
- Enables HTTPS development (`next dev --experimental-https`)
- Supports custom hostnames (e.g., `https://local.example.com:3000`)

**Port resolution priority:**
1. `config.baseUrl` (full override)
2. `config.port` (explicit port)
3. Auto-detected port via `@workflow/utils/get-port`
4. PORT env var
5. Default to 3000

### Breaking Changes

API signature updated:
```typescript
// Before
createEmbeddedWorld(dataDir?: string, port?: number)

// After  
createEmbeddedWorld(args?: Partial<Config>)
```

Closes #233  
Closes #245

### CODE REVIEW

**Comment by pranaygp** [comment]
since #233 was merged, could you please rebase/merge this PR on main? this or includes a number of commits from @adriandlam

File: packages/world-local/src/config.test.ts [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+      expect(result).toBe('https://example.com');
+      expect(getPort).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('edge cases', () => {
+    it('should handle empty config object', async () => {
+      const { getPort } = await import('@workflow/utils/get-port');
+      vi.mocked(getPort).mockResolvedValue(undefined);
```
Comment (VaguelySerious) [inline packages/world-local/src/config.test.ts:232]:
```suggestion
      vi.mocked(getPort).mockResolvedValue({});
```

Comment (aryasaatvik) [reply packages/world-local/src/config.test.ts:232]: fixed the issue. it was overriding defaults with undefined. getPort returns undefined instead of an empty object

**Review by VaguelySerious — APPROVED** [review]

**Comment by VaguelySerious** [comment]
Looks great, thanks again!
